### PR TITLE
Accept optional zkStateManagerVersion in rollup_getZkEVMStateMerkleProofV0

### PR DESCRIPTION
**Pull request by gauravahuja**
## PR Description

- Skip UNSUPPORTED_VERSION when the client omits or sends a blank zkStateManagerVersion. 
- Mark the JSON property as non-required on RollupGetZkEvmStateV0Parameter. 
- Add Jackson parse tests for the parameter object and a method test for a null version.

## Fixed Issue(s)

fixes https://github.com/Consensys/linea-monorepo/issues/2752

## Documentation

- [ ] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [ ] I thought about adding a changelog entry, and added one if I deemed necessary.

---

> [!NOTE]
> **Low Risk**
> Low risk: removes a strict request-parameter validation path and adds tests; behavior change is limited to version handling for a single RPC method.
> 
> **Overview**
> `rollup_getZkEVMStateMerkleProofV0` no longer rejects requests with `UNSUPPORTED_VERSION`; version validation (and the `JsonInvalidVersionMessage` payload type) is removed so omitted/blank versions can proceed.
> 
> The request parameter model marks `zkStateManagerVersion` as optional for Jackson deserialization, and tests are updated/added to cover parsing with missing/null/empty values and the RPC method succeeding when the version is omitted.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 5b62d05851fe75ecb1cdaa40a6f9f281fa47832a. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

**Review by jonesho — APPROVED** [review]

**Review by fluentcrafter — APPROVED** [review]

File: services/rpc/server/src/main/java/net/consensys/shomei/rpc/server/model/RollupGetZkEvmStateV0Parameter.java [hunk 1]
```diff
@@ -27,7 +27,8 @@ public class RollupGetZkEvmStateV0Parameter {
   public RollupGetZkEvmStateV0Parameter(
       @JsonProperty("startBlockNumber") final String startBlockNumber,
       @JsonProperty("endBlockNumber") final String endBlockNumber,
-      @JsonProperty("zkStateManagerVersion") final String zkStateManagerVersion) {
+      @JsonProperty(value = "zkStateManagerVersion", required = false)
```
Comment (Filter94) [inline services/rpc/server/src/main/java/net/consensys/shomei/rpc/server/model/RollupGetZkEvmStateV0Parameter.java:30]: I think we should remove it, we don't ever really need to parse it, right? All we need is not to fail if it's present

Comment (fluentcrafter) [reply services/rpc/server/src/main/java/net/consensys/shomei/rpc/server/model/RollupGetZkEvmStateV0Parameter.java:30]: I agree, it makes sense to me as well.

Comment (gauravahuja) [reply services/rpc/server/src/main/java/net/consensys/shomei/rpc/server/model/RollupGetZkEvmStateV0Parameter.java:30]: Jackson's object mapper is used to parse the json string to the object type. Since the object mapper initialization and configuration is buried under layers of abstraction I cannot be certain if it has FAIL_ON_UNKNOWN_PROPERTIES disabled or not. The best I can do without messing with object mapper config is to configure this parameter as not required.
